src/Psalm/Internal/Type/TypeParser.php

```diff
@@ -420,8 +421,8 @@ public static function getTypeFromTree(
             return new TLiteralFloat((float) $parse_tree->value, $from_docblock);
         }
 
-        if (preg_match('/^\-?(0|[1-9][0-9]*)$/', $parse_tree->value)) {
-            return new TLiteralInt((int) $parse_tree->value, $from_docblock);
+        if (preg_match('/^\-?(0|[1-9]([0-9_]*[0-9])?)$/', $parse_tree->value)) {
```

Comment:
do we want `int<0_000, 1_000>` to work? PHP allows it but I think this regex won't

Reply:
We did not allow leading zeroes, and PHP interprets them as octal numbers, which can get pretty confusing: https://3v4l.org/hprRC, so I don't think we want that.

Reply:
Right, forgot about octals. Fine by me then!